lib/assets/chart_builder/main.js

```diff
@@ -1,358 +1,292 @@
-                <option value="mean">Mean</option>
-                <option value="sum">Sum</option>
-              </select>
+            <div class="group">
+              <BaseSelect
+                name="color_field"
+                label="Color"
+                v-model="fields.color_field"
+                :options="axisOptions"
+                :disabled="noDataVariable"
+              />
+              <BaseSelect
+                name="color_field_type"
+                label="Type"
+                v-model="fields.color_field_type"
+                :options="typeOptions"
+                :disabled="noColorField"
+              />
+              <BaseSelect
+                name="color_field_aggregate"
+                label="Aggregate"
+                v-model="fields.color_field_aggregate"
+                :options="aggregateOptions"
+                :disabled="noColorField"
+              />
             </div>
           </div>
         </div>
+        </form>
       </div>
-    </div>
-  `;
-
-  const channelFields = ["x_field", "y_field", "color_field"];
-  const typeFields = ["x_field_type", "y_field_type", "color_field_type"];
-  const aggregateFields = ["x_field_aggregate", "y_field_aggregate", "color_field_aggregate"];
-
-  const state = {
-    dataOptions: payload.data_options,
-  };
-
-  const selectDataElem = ctx.root.querySelector(`[name="data_variable"]`);
-
-  updateInfoBox(payload.missing_dep);
-
-  if (payload.fresh) {
-    setFreshData(payload.data_options);
-  } else {
-    renderCurrentData(payload.fields, payload.data_options);
-  }
-
-  ctx.root.addEventListener("change", handleFieldChange);
-  selectDataElem.addEventListener("change", updateAxisOptions);
-  channelFieldsListener();
-
-  function handleFieldChange(event) {
-    const { name, value } = event.target;
-    ctx.pushEvent("update_field", { field: name, value });
-  }
-
-  function updateAxisOptions(event) {
-    const { value } = event.target;
-    const axisOptions = state.dataOptions[value];
-    setAxisOptions(axisOptions);
-  }
-
-  function maybeResetFieldOptions(event) {
-    const name = event.target.name;
-    const selectElemType = ctx.root.querySelector(`[name="${name}_type"]`);
-    const selectElemAggregate = ctx.root.querySelector(`[name="${name}_aggregate"]`);
-    const { value } = event.target;
-    if (value && value !== "__count__") {
-      selectElemType.disabled = false;
-      selectElemAggregate.disabled = false;
-    } else {
-      resetField(selectElemType);
-      resetField(selectElemAggregate);
+    `,
+
+    data() {
+      return {
+        fields: payload.fields,
+        dataOptions: payload.data_options,
+        missingDep: payload.missing_dep,
+        chartOptions: ["bar", "line", "point", "area"],
+        typeOptions: ["quantitative", "nominal", "ordinal"],
+        aggregateOptions: ["sum", "mean"],
+        dataVariables: payload.data_options.map(data => data["variable"]),
```

Comment:
```suggestion
        dataVariables: payload.data_options.map(data => data.variable),
```

(either is fine, just saying this is the same :))